docs/Getting Started/Debian/Debian Bullseye Root on ZFS.rst

```diff
@@ -0,0 +1,1118 @@
+deduplication, you will need `massive amounts of RAM
+<http://wiki.freebsd.org/ZFSTuningGuide#Deduplication>`__. Enabling
+deduplication is a permanent change that cannot be easily reverted.
+
+Support
+~~~~~~~
+
+If you need help, reach out to the community using the :ref:`mailing_lists` or IRC at
+`#zfsonlinux <irc://irc.freenode.net/#zfsonlinux>`__ on `freenode
+<https://freenode.net/>`__. If you have a bug report or feature request
+related to this HOWTO, please `file a new issue and mention @rlaager
+<https://github.com/openzfs/openzfs-docs/issues/new?body=@rlaager,%20I%20have%20the%20following%20issue%20with%20the%20Debian%20Bullseye%20Root%20on%20ZFS%20HOWTO:>`__.
+
+Contributing
+~~~~~~~~~~~~
+
+#. Fork and clone: https://github.com/openzfs/openzfs-docs
+
+#. Install the tools::
+
+    sudo apt install python3-pip
+
+    pip3 install -r docs/requirements.txt
+
+    # Add ~/.local/bin to your $PATH, e.g. by adding this to ~/.bashrc:
+    PATH=$HOME/.local/bin:$PATH
+
+#. Make your changes.
+
+#. Test::
+
+    cd docs
+    make html
+    sensible-browser _build/html/index.html
+
+#. ``git commit --signoff`` to a branch, ``git push``, and create a pull
+   request. Mention @rlaager.
+
+Encryption
+~~~~~~~~~~
+
+This guide supports three different encryption options: unencrypted, ZFS
+native encryption, and LUKS. With any option, all ZFS features are fully
+available.
+
+Unencrypted does not encrypt anything, of course. With no encryption
+happening, this option naturally has the best performance.
+
+ZFS native encryption encrypts the data and most metadata in the root
+pool. It does not encrypt dataset or snapshot names or properties. The
+boot pool is not encrypted at all, but it only contains the bootloader,
+kernel, and initrd. (Unless you put a password in ``/etc/fstab``, the
+initrd is unlikely to contain sensitive data.) The system cannot boot
+without the passphrase being entered at the console. Performance is
+good. As the encryption happens in ZFS, even if multiple disks (mirror
+or raidz topologies) are used, the data only has to be encrypted once.
+
+LUKS encrypts almost everything. The only unencrypted data is the bootloader,
+kernel, and initrd. The system cannot boot without the passphrase being
+entered at the console. Performance is good, but LUKS sits underneath ZFS, so
+if multiple disks (mirror or raidz topologies) are used, the data has to be
+encrypted once per disk.
+
+Step 1: Prepare The Install Environment
+---------------------------------------
+
+#. Boot the Debian GNU/Linux Live CD. If prompted, login with the username
+   ``user`` and password ``live``. Connect your system to the Internet as
+   appropriate (e.g. join your WiFi network). Open a terminal.
+
+#. Setup and update the repositories::
+
+     sudo vi /etc/apt/sources.list
+
+   .. code-block:: sourceslist
+
+     deb http://deb.debian.org/debian bullseye main contrib
+     deb http://deb.debian.org/debian bullseye main contrib
```

Comment:
```suggestion
```

Reply:
why not make this https too while we're at it? :)

```suggestion
     deb https://deb.debian.org/debian bullseye main contrib
```

Reply:
Because you cannot use than some apt caching proxies.

Reply:
that seems like a rather edge case... i would rather add security by default than assuming people have a caching service enabled (which is definitely not default)

Reply:
I'm not agree here, because security of debian based on gpg-signing.

Reply:
While I'm ok with http, my comment was about duplicate rows, please remove one.

Reply:
> I'm not agree here, because security of debian based on gpg-signing.

whatever, let's just remove the duplicate lines, i'll send another pr for https.

Reply:
rebased and fixed now.